core/commands/dag/dag.go

```diff
@@ -606,9 +606,16 @@ The output of blocks happens in strict DAG-traversal, first-seen, order.
 
 		// minimal user friendliness
 		if err != nil &&
-			!node.IsOnline &&
 			err == ipld.ErrNotFound {
-			err = fmt.Errorf("%s (currently offline, perhaps retry after attaching to the network)", err)
+			explicitOffline, _ := req.Options["offline"].(bool)
+			if explicitOffline {
+				err = fmt.Errorf("%s (currently offline, perhaps retry without the offline flag)", err)
+			} else {
+				node, envErr := cmdenv.GetNode(env)
```

Comment:
I *believe* if you got here ( `err == ipld.ErrNotFound` ), IPFS already tried GetNode, so that'd be a redundant second call... but I might be wrong.

Otherwise looks great!

Reply:
I swapped out GetNode for GetAPI above. Am I missing something?

Reply:
Ignore what I said, pre-coffee 🤦 
This is good to go!

P.S. node == ipfs object in this context, I misfired node == block